lib/logstorage/values_encoder.go

```diff
@@ -683,56 +684,12 @@ func tryParseIPv4(s string) (uint32, bool) {
 		// Fast path - the entry isn't IPv4
 		return 0, false
 	}
-
-	var octets [4]byte
-	var v uint64
-	var ok bool
-
-	// Parse octet 1
-	n := strings.IndexByte(s, '.')
-	if n <= 0 || n > 3 {
-		return 0, false
-	}
-	v, ok = tryParseDateUint64(s[:n])
-	if !ok || v > 255 {
-		return 0, false
-	}
-	octets[0] = byte(v)
-	s = s[n+1:]
-
-	// Parse octet 2
-	n = strings.IndexByte(s, '.')
-	if n <= 0 || n > 3 {
-		return 0, false
-	}
-	v, ok = tryParseDateUint64(s[:n])
-	if !ok || v > 255 {
+	addr, err := netip.ParseAddr(s)
+	if err != nil || !addr.Is4() {
```

Comment:
Please add benchmarks for negative cases when `tryParseIPv4()` returns `false`. It is important to have a high performance for negative cases as well, since they are located in hot paths of background merge and querying filters.